src/frontend/src/optimizer/plan_node/mod.rs

```diff
@@ -150,13 +153,10 @@ impl ConventionMarker for Stream {
 }
 
 /// The trait for accessing the meta data and [`PlanBase`] for plan nodes.
-pub trait PlanNodeMeta {
-    type Convention: ConventionMarker;
```

Comment:
It was made an associated type because a type cannot be `impl PlanNodeMeta` with different convention marker at the same time.